framework/src/source/rooibos/Test.bs

```diff
@@ -45,47 +54,120 @@ namespace rooibos
     end function
 
     function run()
-      rooibosTimer = createObject("roTimespan")
-      rooibosTimer.mark()
+      m.rooibosTimer = createObject("roTimespan")
 
       if m.isParamTest
-        m.runParamsTest()
+        m.rooibosTimer.mark()
+        promise = m.runParamsTest()
       else
-        m.testSuite[m.funcName]()
+        promise = m.testSuite[m.funcName]()
+      end if
+
+      if rooibos.promises.isPromise(promise)
+        if m.testSuite.isNodeTest
+          m.markDoneWhenTestCompletes(promise)
+        else
+          throw "Can not return a promise from a non-node test"
```

Comment:
Double check that this is a safe way to fail the test and have it show as crashed test.